app/services/channel.rb

```diff
@@ -42,30 +42,22 @@ def channels_for(user, app_name)
       channels.flatten.compact.uniq
     end
 
-    def check_channel_suffix(channel)
-      [STOCK_APP, ADMIN_APP, BROWSE_APP].flatten.
-        any?{ |app_name| channel.ends_with?(app_name) }
-    end
-
-    def add_suffix?(app_name, channel)
-      app_name != DONOR_APP &&
-        channel.starts_with?('user_') &&
-        !check_channel_suffix(channel)
-    end
-
     # TODO: remove this once deleted from PushService
     # add the appropriate app_name suffix on the user channels when registering the device
     # e.g. user_1 becomes user_1_admin, group channels (don't start with 'user_') are unaffected
     # note that donor app channel is just user_1
-
     def add_app_name_suffix(channel_name, app_name)
       [channel_name].flatten.compact.map do |channel|
-        if add_suffix?(app_name, channel)
+        if (!app_name.blank? &&
+             app_name != DONOR_APP &&
+             channel.starts_with?('user_') &&
+             ![STOCK_APP, ADMIN_APP, BROWSE_APP].any?{ |app_name| channel.ends_with?(app_name) })
           "#{channel}_#{app_name}"
```

Comment:
@abhinav098 I've merged this all in to one method because we want to delete it soon anyway. Will protect us from starting to use `add_suffix?` anywhere else in the meantime!

Reply:
:+1: